### PR TITLE
Add support for sslMode key in database config

### LINKED ISSUE #24: The input [sslMode] is invalid
the SSL mode is being provided by clowder in the database config settings, however this field is not supported and the error message

```
The input [sslMode] is invalid
```

Can be seen when attempting to load Clowder configuration that includes this `sslMode` key.

### PR DESCRIPTION
**Pull request by Victoremepunto**
Fixes #24

### CODE REVIEW

**Review by akofink — APPROVED** [review]
Thanks, @Victoremepunto!

File: lib/clowder-common-ruby/types.rb [hunk 1]
```diff
@@ -199,6 +199,7 @@ def valid_keys
         keys << :adminUsername
         keys << :adminPassword
         keys << :rdsCa
+        keys << :sslMode
```
Comment (slemrmartin) [inline lib/clowder-common-ruby/types.rb:202]: when I implemented Clowderization for Sources API few months ago, the sslMode was missing in some cdappconfig.json (I believe in ephemeral environment), just FYI, I haven't checked how it looks now

**Review by slemrmartin — APPROVED** [review]
:+1: LGTM

File: lib/clowder-common-ruby/types.rb (hunk 1 above)
Comment (Victoremepunto) [reply lib/clowder-common-ruby/types.rb:202]: @slemrmartin I can confirm as of today in ephem. clusters they do provide this, set to `disable`

```
oc -n ephemeral-29 rsh compliance-compliance-backend-57457cbfc5-b85n2
sh-4.4$ cat $ACG_CONFIG | jq '.' | grep -i sslmode
    "sslMode": "disable",
```

**Comment by Victoremepunto** [comment]
I'm sorry to say I'll refrain to push fixes to this repository anymore. this PR has been sitting here for almost a month for nothing, and I just found 3 days ago be78b7d2902199f0843a5d2bb43d92f244433bfa is merged with the fix covered in this PR.

**Pull request closed without merging**